paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationInterceptor.kt

```diff
@@ -88,6 +88,14 @@ internal enum class DeferredIntentConfirmationType(val value: String) {
     None("none")
 }
 
+internal class InvalidDeferredIntentUsageException : Exception() {
+    override val message: String = """
+        It appears you are reusing an intent on every `createIntentCallback` call. You should always create a brand
+        new intent in `createIntentCallback` so that the intent has a clean state to start from when attempting to
+        confirm a payment.
```

Comment:
```suggestion
        It appears you are reusing an intent on every `createIntentCallback` call. You should either create a brand
        new intent in `createIntentCallback` or update the existing intent with the new payment method ID.
```

something like this - should include the option to update the intent!